File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
from collections import deque
def solution(n, m, section):
    q = deque(section)
    first = q.popleft()
    answer = 1
    while q:
        c = q.popleft()
        if first+m >c:
            continue
        else:
            first = c
            answer +=1
    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var answer = 0;
    let idx = 1
    
    const sectionArr = Array.from({legnth : n + 1}, () => false)
    for(let s of section){
        sectionArr[s] = true
    }
    
    while(idx <= n){
        if(sectionArr[idx]){
            idx += m
            answer++;
        }else{
            idx++
        }
    }
    
    return answer;
}